Azure.Core.Pole.Tests/TypeSpecGenerator.cs

```diff
@@ -2,19 +2,69 @@
 // Licensed under the MIT License.
 
 using Azure.Core.Pole.Tooling;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using NUnit.Framework;
+using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 
 namespace Azure.Core.Pole.Tests
 {
     public partial class TypeSpecGeneratorTests
     {
         [Test]
-        public void Basics()
+        public void DogModelGeneration()
         {
-            var source = Path.Combine(".", "tsp", "simple.tsp");
-            var destination = Path.Combine(Path.GetTempPath());
+            // Test with Dog model format
+            string source = Path.Combine(".", "tsp", "dog.tsp");
+            string destination = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
             TypeSpecGenerator.Generate(source, destination);
```

Comment:
Add API to the generator to generate sources in-memory without the need to save files to disk. Then rewrite this test such that all the generation and verification happens without writing to disk